neo/SDK/TX/TxManager.cs

```diff
@@ -0,0 +1,198 @@
+
+            var gasBalance = new Nep5API(rpcClient).BalanceOf(NativeContract.GAS.Hash, sender);
+            if (gasBalance >= Tx.SystemFee + Tx.NetworkFee) return this;
+            throw new InvalidOperationException($"Insufficient GAS in address: {sender.ToAddress()}");
+        }
+
+        /// <summary>
+        /// Estimate NetwotkFee, assuming the witnesses are basic Signature Contract
+        /// </summary>
+        private long EstimateNetwotkFee()
+        {
+            long networkFee = 0;
+            UInt160[] hashes = Tx.GetScriptHashesForVerifying(null);
+            int size = Transaction.HeaderSize + Tx.Attributes.GetVarSize() + Tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
+
+            // assume the hashes are single Signature
+            foreach (var hash in hashes)
+            {
+                size += 166;
+                networkFee += ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES64] + ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES33] + InteropService.GetPrice(InteropService.Neo_Crypto_CheckSig, null);
+            }
+
+            networkFee += size * new PolicyAPI(rpcClient).GetFeePerByte();
+            return networkFee;
+        }
+
+        /// <summary>
+        /// Calculate NetworkFee with context items
+        /// </summary>
+        private long CalculateNetworkFee()
+        {
+            long networkFee = 0;
+            UInt160[] hashes = Tx.GetScriptHashesForVerifying(null);
+            int size = Transaction.HeaderSize + Tx.Attributes.GetVarSize() + Tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
+            foreach (UInt160 hash in hashes)
+            {
+                byte[] witness_script = Context.GetScript(hash);
+                if (witness_script is null)
+                {
+                    try
+                    {
+                        witness_script = rpcClient.GetContractState(hash.ToString())?.Script;
+                    }
+                    catch { }
+                }
+
+                if (witness_script is null) continue;
+
+                networkFee += Wallet.CalculateNetWorkFee(witness_script, ref size);
+            }
+            networkFee += size * new PolicyAPI(rpcClient).GetFeePerByte();
+            return networkFee;
+        }
+
+        /// <summary>
+        /// Add Signature
+        /// </summary>
+        public TxManager AddSignature(KeyPair key)
+        {
+            var contract = Contract.CreateSignatureContract(key.PublicKey);
+
+            byte[] signature = Tx.Sign(key);
+            if (!Context.AddSignature(contract, key.PublicKey, signature))
+            {
+                throw new Exception("AddSignature failed!");
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Add Multi-Signature
+        /// </summary>
+        public TxManager AddMultiSig(KeyPair key, params ECPoint[] publicKeys)
```

Comment:
Summary for the arguments in all the SDK

Reply:
I have added the arguments comments, please help to review😄